pkg/controller/direct/cloudbuild/workerpool_controller.go

```diff
@@ -90,7 +91,7 @@ func (m *model) AdapterForObject(ctx context.Context, reader client.Reader, u *u
 	}
 
 	// Get GCP Project
-	projectRef, err := refs.ResolveProject(ctx, reader, obj, obj.Spec.ProjectRef)
+	projectRef, err := resolverefs.ResolveProjectRef(ctx, reader, obj, obj.Spec.ProjectRef)
```

Comment:
So I moved the resolvers into the API package recently.  The intent was that we could then have the resolvers implement an interface, and make a lot of this generic.

WDYT - can we put the resolver functions into the refs package?   (We could even just add a `ResolveRef` function to each ref in this PR?)

Reply:
Regarding the placement of resolver functions, I don't have a strong preference. It could be `api/refs/networkref.go` or `direct/refs/networkref.go`. I think we should maintain consistency in the future. I'm open to moving the resolver functions to `api/refs` if that's preferable.

My main concern here is avoiding placing these functions in individual controller packages, for example `direct/compute/networkref.go`. Some references are used in multiple resources, like `ComputeNetWork`, which is referenced by `CloudBuildWorkerPool` and `ComputeForwardingRule` (and potentially others).